TST: Cron job with Python 3.10.0a0 broken

Example log: https://travis-ci.org/github/astropy/astropy/jobs/694700974

Towards the end:
```
building 'Cython.Plex.Scanners' extension
    creating build/temp.linux-x86_64-3.10
    creating build/temp.linux-x86_64-3.10/tmp
    creating build/temp.linux-x86_64-3.10/tmp/pip-install-u_x07iib
    creating build/temp.linux-x86_64-3.10/tmp/pip-install-u_x07iib/cython
    creating build/temp.linux-x86_64-3.10/tmp/pip-install-u_x07iib/cython/Cython
    creating build/temp.linux-x86_64-3.10/tmp/pip-install-u_x07iib/cython/Cython/Plex
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -I/home/travis/build/astropy/astropy/.tox/.package/include -I/opt/python/3.10-dev/include/python3.10 -c /tmp/pip-install-u_x07iib/cython/Cython/Plex/Scanners.c -o build/temp.linux-x86_64-3.10/tmp/pip-install-u_x07iib/cython/Cython/Plex/Scanners.o
    /tmp/pip-install-u_x07iib/cython/Cython/Plex/Scanners.c: In function â€˜__Pyx_PyList_Appendâ€™:
    /tmp/pip-install-u_x07iib/cython/Cython/Plex/Scanners.c:1395:23: error: lvalue required as left operand of assignment
             Py_SIZE(list) = len+1;
                           ^
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/travis/build/astropy/astropy/.tox/.package/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-u_x07iib/cython/setup.py'"'"'; __file__='"'"'/tmp/pip-install-u_x07iib/cython/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-b3hsghay/install-record.txt --single-version-externally-managed --compile --install-headers /home/travis/build/astropy/astropy/.tox/.package/include/site/python3.10/cython Check the logs for full command output.

=================================== log end ====================================
[31mERROR: could not install deps [setuptools, setuptools_scm, wheel, cython==0.29.14, jinja2==2.10.3, oldest-supported-numpy, extension-helpers]; v = InvocationError('/home/travis/build/astropy/astropy/.tox/.package/bin/python -m pip install setuptools setuptools_scm wheel cython==0.29.14 jinja2==2.10.3 oldest-supported-numpy extension-helpers', 1)
[0m[8124] /home/travis/build/astropy/astropy$ /home/travis/build/astropy/astropy/.tox/.package/bin/python .tox/.tox/lib/python3.10/site-packages/tox/helper/build_requires.py setuptools.build_meta '' >.tox/.package/log/.package-2.log
[31mERROR: invocation failed (exit code 1), logfile: /home/travis/build/astropy/astropy/.tox/.package/log/.package-2.log
[0m================================== log start ===================================
Traceback (most recent call last):
  File "/home/travis/build/astropy/astropy/.tox/.tox/lib/python3.10/site-packages/tox/helper/build_requires.py", line 11, in <module>
    for_build_requires = backend.get_requires_for_build_sdist(None)
  File "/home/travis/build/astropy/astropy/.tox/.package/lib/python3.10/site-packages/setuptools/build_meta.py", line 152, in get_requires_for_build_sdist
    return self._get_build_requires(config_settings, requirements=[])
  File "/home/travis/build/astropy/astropy/.tox/.package/lib/python3.10/site-packages/setuptools/build_meta.py", line 128, in _get_build_requires
    self.run_setup()
  File "/home/travis/build/astropy/astropy/.tox/.package/lib/python3.10/site-packages/setuptools/build_meta.py", line 143, in run_setup
    exec(compile(code, __file__, 'exec'), locals())
  File "setup.py", line 11, in <module>
    from extension_helpers import get_extensions
ModuleNotFoundError: No module named 'extension_helpers'
```

Is this problem on our end? If not, how useful is the python-dev job really?

--- a/.travis.yml
+++ b/.travis.yml
@@ -171,7 +171,7 @@ matrix:
 
         # Test against Python dev in cron job.
         - language: python
-          python: nightly
+          python: 3.9-dev
           dist: bionic
           name: Python dev with basic dependencies
           stage: Cron tests

